tests/Transports.AspNetCore.Tests/NewtonsoftJsonTests.cs

```diff
@@ -0,0 +1,86 @@
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Server.Transports.AspNetCore.Tests
+{
+    public partial class NewtonsoftJsonTests
+    {
+        [Fact]
+        public async Task Decodes_Request()
```

Comment:
Need a test with complex variables/extensions.

Reply:
Done for both nested dictionary and array